pkg/asset/installconfig/gcp/client.go

```diff
@@ -0,0 +1,74 @@
+package gcp
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+// API represents the calls made to the API.
+type API interface {
+	GetNetwork(ctx context.Context, network, project string) (*compute.Network, error)
+	GetSubnetworks(ctx context.Context, network, project, region string) ([]*compute.Subnetwork, error)
+}
+
+// Client makes calls to the GCP API.
+type Client struct{}
+
+// GetNetwork uses the GCP Compute Service API to get a network by name from a project.
+func (c *Client) GetNetwork(ctx context.Context, network, project string) (*compute.Network, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+	res, err := svc.Networks.Get(project, network).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get network %s", network)
+	}
+	return res, nil
+}
+
+// GetSubnetworks uses the GCP Compute Service API to retrieve all subnetworks in a given network.
+func (c *Client) GetSubnetworks(ctx context.Context, network, project, region string) ([]*compute.Subnetwork, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := svc.Subnetworks.List(project, region)
+	var res []*compute.Subnetwork
+	if err := req.Pages(ctx, func(page *compute.SubnetworkList) error {
+		for _, subnet := range page.Items {
+			if strings.Contains(subnet.Network, network) {
+				res = append(res, subnet)
+			}
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func getComputeService(ctx context.Context) (*compute.Service, error) {
+	ssn, err := GetSession(ctx)
```

Comment:
It would be nice to keep the session in the client.

Reply:
I like this idea and have been looking into it. It's going to take a bit more work because we will have to adapt dns (the only place session is used) to use client, which I was planning on doing anyway. I also think we will need to not pass the context to GetSession but originate context inside GetSession, otherwise it gets too clunky. I will try to write this into this PR.

Reply:
A function, `NewClient(ctx)` -> can call the `GetSession(ctx)` and store the `session.Credentials` that can be used later on.

The contecxt for gettting session.Credentials and the actual `GetNetwork` can therefore be different.